File: ShippingMark/Models/ChiTiet.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
namespace ShippingMark.Models
{    
    public class ChiTiet
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int ID { get; set; }
        [Display(Name = "REFNO Đơn hàng")]
        public string REFNO { get; set; }
        [Display(Name = "PoNo Số PO")]
        public string PoNo { get; set; }
        [Display(Name = "MODEL")]
        public string MODEL { get; set; }
        [Display(Name = "SIZE Kích thước")]
        public string SIZE { get; set; }
        // Thêm các trường còn lại theo cấu trúc bảng
        [Display(Name = "Net weight")]
        public string NW { get; set; }
        [Display(Name = "Gross weight")]
        public string GW { get; set; }        
        [Display(Name = "Dài (m)")]
        public decimal? Dai { get; set; }

        [Display(Name = "Rộng (m)")]
        public decimal? Rong { get; set; }
        [Display(Name = "Cao (m)")]
        public decimal? Cao { get; set; }
        
        [Display(Name = "Màu (Color)")]
        public string COLOR { get; set; }

        public string Type { get; set; }
        [Display(Name = "Loại đóng gói")]
        public TypeNum SelectedTypeNum { get; set; }

        [Display(Name = "Nội dung mở rộng")]
        public string TypeSuffix { get; set; }
        [Display(Name = "In")]
        public bool? usingPrint { get; set; }
        [Display(Name = "Bắt đầu")]
        public string StartSerial { get; set; }
        [Display(Name = "Số lượng")]
     
        public string NumSerial { get; set; }
        [Display(Name = "Lô")]
        public string LotNo { get; set; }
        [Display(Name = "Khách hàng")]
        public string Customer { get; set; }
        public string QUANTITY { get; set; }
        public string COMMODITY { get; set; }
        public string CustomerCode { get; set; }

        // Ghi nhận thông tin user
        public string CreatedBy { get; set; }
        [Column(TypeName = "datetime2")]
        public DateTime CreatedDate { get; set; }
        public string LastModifiedBy { get; set; }
        [Column(TypeName = "datetime2")]
        public DateTime? LastModifiedDate { get; set; }
    }
}

File: app/Models/ChiTiet.php
<?php

namespace App\Models;

use App\Enums\TypeNum;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class ChiTiet extends Model
{
    use HasFactory;

    protected $table = 'chi_tiets';

    protected $fillable = [
        'ref_no',
        'po_no',
        'model',
        'size',
        'nw',
        'gw',
        'dai',
        'rong',
        'cao',
        'color',
        'type',
        'selected_type_num',
        'type_suffix',
        'using_print',
        'start_serial',
        'num_serial',
        'lot_no',
        'customer',
        'quantity',
        'commodity',
        'customer_code',
        'created_by',
        'last_modified_by',
    ];

    protected $casts = [
        'selected_type_num' => TypeNum::class,
        'using_print' => 'boolean',
        'dai' => 'decimal:3',
        'rong' => 'decimal:3',
        'cao' => 'decimal:3',
    ];
}
